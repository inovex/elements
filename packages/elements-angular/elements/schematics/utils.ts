import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
