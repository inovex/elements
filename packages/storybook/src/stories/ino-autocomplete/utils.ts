import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
