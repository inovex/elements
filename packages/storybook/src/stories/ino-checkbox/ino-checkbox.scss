.story-checkbox {
  ino-checkbox {
    width: 33%;
  }

  .story-checkbox-states {
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    align-items: center;
    margin: auto;
    text-align: center;
  }

  .story-selection-states {
    display: flex;
    justify-content: space-around;
    align-self: center;
    text-align: center;
  }

  .story-selection-states ino-checkbox {
    width: 25%;
  }

  .customizable-checkbox {
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 40px;
    align-self: center;
  }
}
