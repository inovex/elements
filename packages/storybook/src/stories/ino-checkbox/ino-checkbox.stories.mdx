import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-checkbox/readme.md";

<Meta title="Input/ino-checkbox/Documentation" />

<Description markdown={readme} />
