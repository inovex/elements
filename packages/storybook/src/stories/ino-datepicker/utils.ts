import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
