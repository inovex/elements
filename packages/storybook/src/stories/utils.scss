/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
