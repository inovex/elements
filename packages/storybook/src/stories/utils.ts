import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
