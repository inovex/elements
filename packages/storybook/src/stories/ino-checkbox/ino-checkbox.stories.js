import { boolean, text } from '@storybook/addon-knobs';

import componentReadme from '_local-elements/src/components/ino-checkbox/readme.md';
import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import './ino-checkbox.scss';

// https://github.com/storybooks/storybook/issues/4337#issuecomment-428495664
function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function (e) {
    const el = e.target;
    if (el.tagName.toLowerCase() !== 'ino-checkbox') {
      return;
    }

    el.setAttribute('checked', e.detail);

    if (el.getAttribute('indeterminate') === 'true') {
      el.setAttribute('indeterminate', 'false');
    }
  };

  document.addEventListener('checkedChange', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('checkedChange', eventHandler);
  };
}

export default {
  title: 'Input/<ino-checkbox>',
  parameters: {
    actions: {
      handles: ['checkedChange .customizable-checkbox']
    }
  },
  decorators: [
    withStencilReadme(componentReadme),
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

export const DefaultUsage = () => /*html*/ `
<div class="story-checkbox">
  <ino-checkbox
    class="customizable-checkbox"
    name="customizable-checkbox"
    value="placeholder-value"
    checked="${boolean('checked', false)}"
    disabled="${boolean('disabled', false)}"
    indeterminate="${boolean('indeterminate', false)}"
    ino-selection="${boolean('ino-selection', false)}"
  >
    ${text('<slot />', 'Customizable checkbox')}
  </ino-checkbox>
  <h4>Default</h4>
  <div class="story-checkbox-states">
    <ino-checkbox checked>Checked</ino-checkbox>
    <ino-checkbox>Unchecked</ino-checkbox>
    <ino-checkbox indeterminate>Indeterminated</ino-checkbox>
    <ino-checkbox disabled checked>Checked and Disabled</ino-checkbox>
    <ino-checkbox disabled>Unchecked and Disabled</ino-checkbox>
    <ino-checkbox disabled indeterminate>Indeterminated and Disabled</ino-checkbox>
  </div>

  <h4>Selection</h4>
  <div class="story-selection-states">
       <ino-checkbox ino-selection checked>Checked</ino-checkbox>
       <ino-checkbox ino-selection>Unchecked</ino-checkbox>
       <ino-checkbox checked ino-selection disabled>Checked and disabled</ino-checkbox>
       <ino-checkbox ino-selection disabled>Unchecked and disabled</ino-checkbox>
  </div>
</div>
`;

DefaultUsage.storyName = 'Default usage';
