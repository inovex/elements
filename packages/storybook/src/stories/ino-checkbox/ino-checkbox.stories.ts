import { Components } from '@inovex.de/elements';
import { useEffect } from '@storybook/client-api';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';

import { defaultDecorator } from '../utils';
import './ino-checkbox.scss';

export default {
  title: 'Input/ino-checkbox',
  component: 'ino-checkbox',
  decorators: [
    defaultDecorator,
    (story) => {
      useEffect(() => {
        const handleCheckedChange = (e) => {
          const checkbox: HTMLInoCheckboxElement = e.target;
          checkbox.checked = e.detail;
          if (checkbox.indeterminate) {
            checkbox.indeterminate = false;
          }
        };

        const checkboxes = document.querySelectorAll('ino-checkbox');
        checkboxes.forEach((c) =>
          c.addEventListener('checkedChange', handleCheckedChange)
        );
        return () =>
          checkboxes.forEach((c) =>
            c.removeEventListener('checkedChange', handleCheckedChange)
          );
      });
      return story();
    },
  ],
};

export const Playground: Story<Components.InoCheckbox> = (args) => html`
  <ino-checkbox
    class="customizable-checkbox"
    checked="${args.checked}"
    disabled="${args.disabled}"
    indeterminate="${args.indeterminate}"
    name="${args.name}"
    selection="${args.selection}"
    value="${args.value}"
  >
    Label
  </ino-checkbox>
`;

Playground.args = {
  checked: false,
  disabled: false,
  indeterminate: false,
  name: '',
  selection: false,
  value: '',
};

export const Variations = () => html`
  <div class="story-states">
    <ino-checkbox checked>Checked</ino-checkbox>
    <ino-checkbox>Unchecked</ino-checkbox>
    <ino-checkbox indeterminate>Indeterminated</ino-checkbox>
  </div>
  <div class="story-states">
    <ino-checkbox disabled checked>Checked and Disabled</ino-checkbox>
    <ino-checkbox disabled>Unchecked and Disabled</ino-checkbox>
    <ino-checkbox disabled indeterminate
      >Indeterminate and Disabled
    </ino-checkbox>
  </div>
`;

export const Selection = () => html`
  <div class="story-states">
    <ino-checkbox selection checked>Checked</ino-checkbox>
    <ino-checkbox selection>Unchecked</ino-checkbox>
    <ino-checkbox selection indeterminate>Indeterminate</ino-checkbox>
  </div>
  <div class="story-states">
    <ino-checkbox checked selection disabled
      >Checked and disabled
    </ino-checkbox>
    <ino-checkbox selection disabled>Unchecked and disabled</ino-checkbox>
    <ino-checkbox selection indeterminate disabled>
      Indeterminate Disabled
    </ino-checkbox>
  </div>
`;
