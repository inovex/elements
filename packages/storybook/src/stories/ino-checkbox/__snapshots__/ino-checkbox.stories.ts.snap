// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input/ino-checkbox Checked smoke-test 1`] = `
<div id="root-inner">
  <div class="story-checkbox">
    <ino-checkbox checked="true"
                  disabled="false"
                  indeterminate="false"
                  name="my-name"
                  selection="false"
                  value="my-value"
                  class="hydrated"
    >
      Label
      <input type="hidden"
             class="aux-input"
             name="my-name"
             value="my-value"
      >
    </ino-checkbox>
  </div>
</div>
`;

exports[`Input/ino-checkbox Disabled smoke-test 1`] = `
<div id="root-inner">
  <div class="story-checkbox">
    <ino-checkbox checked="true"
                  disabled="true"
                  indeterminate="false"
                  name="my-name"
                  selection="false"
                  value="my-value"
                  class="hydrated"
    >
      Label
      <input type="hidden"
             class="aux-input"
             disabled
             name="my-name"
             value="my-value"
      >
    </ino-checkbox>
  </div>
</div>
`;

exports[`Input/ino-checkbox Indeterminate smoke-test 1`] = `
<div id="root-inner">
  <div class="story-checkbox">
    <ino-checkbox checked="true"
                  disabled="false"
                  indeterminate="true"
                  name="my-name"
                  selection="false"
                  value="my-value"
                  class="hydrated"
    >
      Label
      <input type="hidden"
             class="aux-input"
             name="my-name"
             value="my-value"
      >
    </ino-checkbox>
  </div>
</div>
`;

exports[`Input/ino-checkbox Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-checkbox">
    <ino-checkbox checked="true"
                  disabled="false"
                  indeterminate="false"
                  name="my-name"
                  selection="false"
                  value="my-value"
                  class="hydrated"
    >
      Label
      <input type="hidden"
             class="aux-input"
             name="my-name"
             value="my-value"
      >
    </ino-checkbox>
  </div>
</div>
`;

exports[`Input/ino-checkbox Selection smoke-test 1`] = `
<div id="root-inner">
  <div class="story-checkbox">
    <ino-checkbox checked="true"
                  disabled="false"
                  indeterminate="false"
                  name="my-name"
                  selection="true"
                  value="my-value"
                  class="hydrated"
    >
      Label
      <input type="hidden"
             class="aux-input"
             name="my-name"
             value="my-value"
      >
    </ino-checkbox>
  </div>
</div>
`;
